File: src/main/java/at/hrechny/predictionsbot/Application.java
package at.hrechny.predictionsbot;

import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Factory;
import io.micronaut.runtime.Micronaut;
import java.time.Clock;

public class Application {

  public static void main(String[] args) {
    Micronaut.run(Application.class, args);
  }

  @Factory
  static class ApplicationFactory {

    @Bean
    Clock clock() {
      return Clock.systemUTC();
    }

  }
}

File: src/main/kotlin/at/hrechny/predictionsbot/Application.kt
package at.hrechny.predictionsbot

import io.micronaut.context.annotation.Bean
import io.micronaut.context.annotation.Factory
import io.micronaut.runtime.Micronaut
import java.time.Clock

class Application {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            Micronaut.run(Application::class.java, *args)
        }
    }

    @Factory
    class ApplicationFactory {
        @Bean
        fun clock(): Clock = Clock.systemUTC()
    }
}
